src/gwt/src/org/rstudio/studio/client/workbench/views/vcs/common/diff/UnifiedEmitter.java

```diff
@@ -353,18 +354,18 @@ private int writeDiffLine(ArrayList<Line> output, Line dff, int skew)
          switch (dff.getType())
          {
             case Deletion:
+               skew--;
```

Comment:
This (and below) is really the meat of the change; the rest is mostly just debugging + slight reformatting. This two-line change took me most of the day to divine!